pkg/controller/service.go

```diff
@@ -180,20 +199,25 @@ func (s *ShadowServiceManager) cleanupPortMapping(oldUserSvc *corev1.Service, ne
 			}
 		}
 	}
+
+	return nil
 }
 
-func (s *ShadowServiceManager) getShadowServicePorts(svc *corev1.Service) []corev1.ServicePort {
+func (s *ShadowServiceManager) getShadowServicePorts(svc *corev1.Service) ([]corev1.ServicePort, error) {
 	var ports []corev1.ServicePort
 
-	svcMode := s.getServiceMode(svc)
+	trafficType, err := annotations.GetTrafficType(s.defaultTrafficType, svc.Annotations)
+	if err != nil {
+		return []corev1.ServicePort{}, fmt.Errorf("unable to get service traffic-type: %w", err)
+	}
 
 	for i, sp := range svc.Spec.Ports {
-		if !(svcMode == k8s.ServiceTypeUDP && sp.Protocol == corev1.ProtocolUDP) && !(svcMode != k8s.ServiceTypeUDP && sp.Protocol == corev1.ProtocolTCP) {
+		if !(trafficType == annotations.ServiceTypeUDP && sp.Protocol == corev1.ProtocolUDP) && !(trafficType != annotations.ServiceTypeUDP && sp.Protocol == corev1.ProtocolTCP) {
```

Comment:
Is there a way to clean this up to make it more readable, or move to a function?

Reply:
perhaps a switch with the different commented cases? just because there are now a lot of not comparisons